### PR TITLE
Add buffer types to struct module.

### PR DESCRIPTION
**Pull request by euresti**
I noticed these were missing and I made the version look alike for an easy rename into 2and3.

### CODE REVIEW

File: stdlib/3/struct.pyi [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
+def calcsize(fmt: _FmtType) -> int: ...
 
 class Struct:
-    format = b''
-    size = 0
+    format = ...  # type: bytes
+    size = ...  # type: int
 
-    def __init__(self, format: str) -> None: ...
+    def __init__(self, format: Union[bytes, Text]) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/struct.pyi]: You can use `_FmtType` here too.

File: stdlib/3/struct.pyi [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
+from typing import Any, Tuple, Text, Union
 
 class error(Exception): ...
 
-def pack(fmt: str, *v: Any) -> bytes: ...
-# TODO buffer type
-def pack_into(fmt: str, buffer: Any, offset: int, *v: Any) -> None: ...
+_FmtType = Union[bytes, Text]
+if sys.version_info >= (3,):
+    _BufferType = Union[bytes, bytearray, memoryview]
```
Comment (JelleZijlstra) [inline stdlib/3/struct.pyi]: There's a unit test checking that `array.array` is also accepted (http://bugs.python.org/issue1563759). Other objects that implement the C-level buffer protocol should also be affected, but I'm not sure how we can express that in typeshed.

**Comment by JelleZijlstra** [comment]
Thanks! We're still missing `iter_unpack`, which was added in 3.4. Feel free to add that in this PR.

File: stdlib/3/struct.pyi (hunk 2 above)
Comment (euresti) [inline stdlib/3/struct.pyi]: Added array.  I noticed that in python 2 array is a `Generic[T]` however manually testing you can use anything in the T (e.g. array[int] works perfectly fine) so I left it as array. Should it be `array[Any]`?

Comment (JelleZijlstra) [inline stdlib/3/struct.pyi]: Yes, I'd prefer `array[Any]` for explicitness; at some point I'd like to make sure typeshed doesn't use any generic types without type parameters.

**Comment by JelleZijlstra** [comment]
Thanks!

**Pull request merged**